xcb.c

```diff
@@ -170,6 +172,10 @@ void grab_pointer_and_keyboard(xcb_connection_t *conn, xcb_screen_t *screen, xcb
 
     int tries = 10000;
 
+    /* Using few variables to trigger a redraw_screen() if too many tries */
+    bool redrawn = false;
+    time_t start = clock(), end;
```

Comment:
I understand it can seem like some kind of abstruse syntactic sugar, I'll change that indeed.